passes/cmds/plugin.cc

```diff
@@ -36,14 +44,48 @@ void load_plugin(std::string filename, std::vector<std::string> aliases)
 	if (filename.find('/') == std::string::npos)
 		filename = "./" + filename;
 
+	#ifdef WITH_PYTHON
+	if (!loaded_plugins.count(filename) && !loaded_python_plugins.count(filename)) {
+	#else
 	if (!loaded_plugins.count(filename)) {
+	#endif
+
+		#ifdef WITH_PYTHON
+		if(boost::algorithm::ends_with(filename, ".py"))
+		{
+			int last_slash = filename.find_last_of('/');
+			std::string path = filename.substr(0, last_slash);
+			filename = filename.substr(last_slash+1, filename.size());
+			filename = filename.substr(0,filename.size()-3);
+			PyRun_SimpleString(("sys.path.insert(0,\""+path+"\")").c_str()); 
+			PyObject *filename_p = PyUnicode_FromString(filename.c_str());
+			if(filename_p == NULL)
+			{
+				log_cmd_error("Issues converting `%s' to Python\n", filename.c_str());
```

Comment:
Is it possible to get better error information through exceptions etc either here or in the PyImport_Import returning null case?

Reply:
I found out that Boost.Python actually provides a function that prints the error to stderr and used that one bbfb43006d2a7d67d06ee151a1b8ad05ec5b1750.
It would be better to write it to Yosys's error log though, but I could not find how to get to the error message itself. The only way I could think of would be to temporarily redirect stderr to a buffer and then write that to the Yosys log, do you have a better idea perhaps?

Reply:
Does this help? https://stackoverflow.com/questions/24386818/how-can-i-get-output-of-pyerr-print-in-windows-or-save-it-as-a-string

Reply:
That's what I found, too. But this solution relies on a function throwing an actual exception. PyUnicode_FromString and PyImport_Import do not throw any exceptions, in case of an error NULL is returned and the user may call PyErr_Print(). All other solutions that I found suggest redirecting stdout and stderr.

Reply:
Ah, I see. This seems fine then.